src/content-helper/editor-sidebar/related-posts/component.tsx

```diff
@@ -293,94 +198,43 @@ export const RelatedPostsPanel = (): React.JSX.Element => {
 				} );
 		};
 
-		const filterTypeIsAuthor = PostFilterType.Author === filter.type;
-		const filterTypeIsTag = PostFilterType.Tag === filter.type;
-		const filterTypeIsSection = PostFilterType.Section === filter.type;
-		const filterTypeIsUnavailable = PostFilterType.Unavailable === filter.type;
-		const noAuthorsExist = 0 === postData.authors.length;
-		const noTagsExist = 0 === postData.tags.length;
-		const noCategoriesExist = 0 === postData.categories.length;
-		const authorIsUnavailable = filterTypeIsAuthor && ! postData.authors.includes( filter.value );
-		const tagIsUnavailable = filterTypeIsTag && ! postData.tags.includes( filter.value );
-		const sectionIsUnavailable = filterTypeIsSection && ! postData.categories.includes( filter.value );
-
 		setLoading( true );
-		if ( filterTypeIsUnavailable || ( filterTypeIsTag && noTagsExist ) ||
-			( filterTypeIsSection && noCategoriesExist ) || ( filterTypeIsAuthor && noAuthorsExist )
-		) {
-			if ( ! isPostDataEmpty() ) {
-				setFilter( getInitialFilterSettings() );
-			}
-		} else if ( tagIsUnavailable ) {
-			setFilter( { type: PostFilterType.Tag, value: postData.tags[ 0 ] } );
-		} else if ( sectionIsUnavailable ) {
-			setFilter( { type: PostFilterType.Section, value: postData.categories[ 0 ] } );
-		} else if ( authorIsUnavailable ) {
-			setFilter( { type: PostFilterType.Author, value: postData.authors[ 0 ] } );
-		}	else {
-			fetchPosts( FETCH_RETRIES );
-		}
+		fetchPosts( FETCH_RETRIES );
 
 		return (): void => {
 			setLoading( false );
 			setPosts( [] );
-			setMessage( '' );
 			setError( undefined );
 		};
-	}, [ period, metric, filter, postData ] );
+	}, [ period, metric, filters, postData ] );
 
 	/**
-	 * Updates the filter value.
-	 *
-	 * @param {string} newFilterValue The new filter value.
+	 * Updates the filters value.
 	 *
 	 * @since 3.11.0
+	 *
+	 * @param {PostFilters}    newValue   The new filters value.
+	 * @param {PostFilterType} filterType The type of filter being changed.
 	 */
-	const updateFilterValue = (
-		newFilterValue: string | null | undefined
+	const updateFilters = (
+		newValue: string | null | undefined,
+		filterType: PostFilterType
 	): void => {
-		if ( typeof newFilterValue === 'string' ) {
-			onFilterChange( filter.type, newFilterValue );
-			setFilter( { ...filter, value: newFilterValue } );
+		if ( newValue === null || newValue === undefined ) {
```

Comment:
Should we be using yoda conditions here? 

```suggestion
		if ( null === newValue || undefined === newValue ) {
```

Reply:
Good call! I've committed this.